packages/visx-demo/src/components/GalleryTile.tsx

```diff
@@ -18,6 +18,43 @@ type Props<ExampleProps extends WidthAndHeight> = {
 const renderLinkWrapper = (url: string | undefined, node: React.ReactNode) =>
   url ? <Link href={url}>{node}</Link> : node;
 
+/**
+ * hook which returns if the ref was ever visible.
+ * used for better perf/not rendering all tiles on load.
+ */
+function useEverVisible() {
+  const ref = useRef();
+  const [everVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          setIsVisible((visible) => visible || entry.isIntersecting);
+        });
+      },
+      {
+        root: null, // viewport is the root
+        threshold: 0.01,
+      },
+    );
+
+    let curr;
+    if (ref.current) {
+      curr = ref.current;
+      observer.observe(ref.current);
+    }
+
+    return () => {
+      if (curr) {
+        observer.unobserve(curr);
+      }
+    };
+  }, [ref]);
```

Comment:
Oh, might not want to use a ref as an effect dep. It'll be stable across re-renders and not trigger the effect. https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node